### ISSUE TITLE
Warning message appears when users set the ‘Task status’ of a ‘Case Investigation’ task to 'Done', if the associated case has not yet been classified

### ISSUE THREAD

**Issue by adinaflorea9** [SORMAS-Foundation/SORMAS-Project#8091]
### Feature Description

1. In the ‘Cases’ directory, on a case that has the ‘Case classification’ set to ‘Not yet classified’, if the user edits the ‘Case investigation’ task and sets its ‘Task Status’ to 'Done' and saves the task, a warning message is displayed:

2. If the user goes on the case page of that same case, and they manually set the case ‘Investigation status’ to ‘Investigation done’ and saves the page, there is no warning message. 

   If the user then goes to open the ‘Case investigation’ task, the ‘Task Status’ is set to ‘Done’ and it is highlighted in red; if they try to save the task the warning message appears, but it has no ‘impact’ since the task status has already been saved when the case page was saved.

3. If the user goes into the 'Bulk Edit Mode' in the 'Tasks' directory, selects 'Edit' from the 'Bulk actions' dropdown and uses this specific task edit form, they can set the 'Task Status' to 'Done' for tasks associated with cases that are still 'Not yet classified' and no warning message is displayed.

   If afterwards the user opens the respective tasks, the ‘Task Status’ is set to ‘Done’ and it is highlighted in red; if they try to save the task the warning message appears, but it has no ‘impact’ since the task status has already been saved.

### Problem Description

1.	Should users be able to set the case ‘Investigation Status’ to ‘Investigation Done’ manually, without first updating the ‘Case Classification’ into something other than ‘Not yet classified’?
     1.1.	If, no, then shouldn’t a warning message be displayed (as it currently is when they try to change the ‘Task Status’)?
     1.2.	If, yes, then the warning message, that appears when they try to change the ‘Task Status’ to ‘Done’ for an unclassified case, should not appear?
2.	If the warning message should appear in the instances described above, then shouldn’t it also appear when changing the ‘Task Status’ to ‘Done’ via the ‘Bulk Edit mode’?

### Additional Information

The reason why I went through these flows is because the warning message that users get when trying to change the 'Task Status' to 'Done' is quite generic, and it says that 'It's not allowed to set **investigation status** to done for an unclassified case'; but it does not appear when users actually change the 'Investigation Status' - 'Done', rather it appears when they change the 'Task Status' - 'Done'.

![image](https://user-images.githubusercontent.com/96909231/155154526-747ef0b4-b9cc-4e32-859e-f92b2a7892fe.png)

**Comment by MateStrysewske** [comment]
@bernardsilenou @Chinedar @foyiri I think it's a problem that the logic is inconsistent here. However, @kwa20 and I believe that also adding validation to the case form would not be a good idea, especially in a situation like the one we have right now where people might not even classify cases because there's just too much stuff going on.

We'd vote for removing the validation from tasks and allowing users to set investigation tasks to done even if the associated cases are not classified. Is that fine for you?

**Comment by Chinedar** [comment]
Yes I vote we remove it. We might later on reconsider bringing it
back...but given the pressure now, I say we remove it.

**Comment by bernardsilenou** [comment]
@MateStrysewske @Chinedar 
- I think the issue here is that the warning message when task status is set to done is talking about investigation status and case classification.
- The goal of case investigation task type is to investigate the case ie setting `case.investigationstatus != pending`, so the validation is to prevent users from setting `task;taskstaus = done` when  `case.investigationstatus = pending`
- If I can remember well, the aim of this validation was to prevent the mobile users from setting investigation tasks to done without first investigating the case (ie classify case, assign epid number, set investigation status to done, etc)
- Maybe we can just correct the warning message: 
   - error message to replace the old one `it is not allowed to set the task status to done when the case investigation status is pending`
   - for bulk editing of tasks
           - if the case investigation status is pending, then we can show  the same warning message

**Comment by MateStrysewske** [comment]
@bernardsilenou Changing the message will not fix the consistency issue that I described in my comment above.

**Comment by bernardsilenou** [comment]
@MateStrysewske @Chinedar 
- I had a look again and I think the logic is fine (ie based on case.investigationstatus) but the error message is not fine. Please can you test also?
- the case investigation task is to inform users to investigate the case, so why allow the users to confirm that the task is done when the case is still uninvestigated?  We want to prevent having pending cases with investigation tasks that are done. If this is not that aim, then can we not just remove this task type from the system?

- what about adding a feature configuration for it, since this task type is applicable to African users only?
